cmd/conduit/root/root.go

```diff
@@ -0,0 +1,120 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package root
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/conduitio/conduit/cmd/conduit/root/pipelines"
+	"github.com/conduitio/conduit/pkg/conduit"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
+	"github.com/conduitio/ecdysis"
+)
+
+var (
+	_ ecdysis.CommandWithFlags         = (*RootCommand)(nil)
+	_ ecdysis.CommandWithExecute       = (*RootCommand)(nil)
+	_ ecdysis.CommandWithDocs          = (*RootCommand)(nil)
+	_ ecdysis.CommandWithSubCommands   = (*RootCommand)(nil)
+	_ ecdysis.CommandWithConfiguration = (*RootCommand)(nil)
+)
+
+type RootFlags struct {
+	Version bool `long:"version" short:"v" usage:"show current Conduit version" persistent:"true"`
+	conduit.Config
+}
+
+type RootCommand struct {
+	flags RootFlags
+	cfg   conduit.Config
+}
+
+func (c *RootCommand) Execute(_ context.Context) error {
+	if c.flags.Version {
+		_, _ = fmt.Fprintf(os.Stdout, "%s\n", conduit.Version(true))
+		return nil
+	}
+
+	e := &conduit.Entrypoint{}
+	e.Serve(c.cfg)
+	return nil
+}
+
+func (c *RootCommand) ParseConfig() ecdysis.Config {
+	path := filepath.Dir(c.flags.ConduitCfgPath)
+	c.cfg = conduit.DefaultConfigWithBasePath(path)
+
+	return ecdysis.Config{
+		EnvPrefix:  "CONDUIT",
+		ParsedCfg:  &c.cfg,
+		ConfigPath: c.flags.ConduitCfgPath,
+		DefaultCfg: c.cfg,
+	}
+}
+
+func (c *RootCommand) Usage() string { return "conduit" }
+
+func (c *RootCommand) Flags() []ecdysis.Flag {
+	flags := ecdysis.BuildFlags(&c.flags)
+
+	currentPath, err := os.Getwd()
+	if err != nil {
+		panic(cerrors.Errorf("failed to get current working directory: %w", err))
+	}
+
+	c.cfg = conduit.DefaultConfigWithBasePath(currentPath)
+	flags.SetDefault("config.path", c.cfg.ConduitCfgPath)
```

Comment:
@hariso also, because otherwise, the flag values will be indeed present (empty) and will take precedence over anything that was previously set via viper (default, config, env) 🤦